modules/reactor/src/meshgenerators/AssemblyMeshGenerator.C

```diff
@@ -487,6 +495,58 @@ AssemblyMeshGenerator::generateMetadata()
     printReactorMetadata("assembly", name());
 }
 
+void
+AssemblyMeshGenerator::generateFlexibleAssemblyBoundaries()
+{
+  // Assemblies that invoke this method have constituent pin lattice, delete outermost background or
+  // duct region (if present)
+  SubdomainName block_to_delete = "";
+  if (_background_region_id.size() == 0)
+    mooseError("Attempting to use flexible stitching on assembly " + name() +
+               " that does not have a background region. This is not yet supported.");
+  const auto radial_index = _duct_region_ids.size() == 0 ? 0 : _duct_region_ids[0].size();
+  block_to_delete =
+      "RGMB_ASSEMBLY" + std::to_string(_assembly_type) + "_R" + std::to_string(radial_index);
+
+  {
+    // Invoke BlockDeletionGenerator to delete outermost mesh interval of assembly
+    auto params = _app.getFactory().getValidParams("BlockDeletionGenerator");
+
+    params.set<std::vector<SubdomainName>>("block") = {block_to_delete};
+    params.set<MeshGeneratorName>("input") = name() + "_delbds";
+
+    addMeshSubgenerator("BlockDeletionGenerator", name() + "_del_outer", params);
+  }
+  {
+    // Invoke FlexiblePatternGenerator to triangulate deleted mesh interval
+    auto params = _app.getFactory().getValidParams("FlexiblePatternGenerator");
+
+    params.set<std::vector<MeshGeneratorName>>("inputs") = {name() + "_del_outer"};
+    params.set<std::vector<libMesh::Point>>("extra_positions") = {libMesh::Point(0, 0, 0)};
+    params.set<std::vector<unsigned int>>("extra_positions_mg_indices") = {0};
+    params.set<bool>("use_auto_area_func") = true;
+    params.set<MooseEnum>("boundary_type") = (_geom_type == "Hex") ? "HEXAGON" : "CARTESIAN";
+    params.set<unsigned int>("boundary_sectors") =
+        getReactorParam<unsigned int>(RGMB::num_sectors_flexible_stitching);
+    params.set<double>("boundary_size") = getReactorParam<Real>(RGMB::assembly_pitch);
+    params.set<boundary_id_type>("external_boundary_id") = _assembly_boundary_id;
+    params.set<std::string>("external_boundary_name") = _assembly_boundary_name;
+    params.set<SubdomainName>("background_subdomain_name") = block_to_delete + "_TRI";
+    params.set<unsigned short>("background_subdomain_id") = 19999;
```

Comment:
same here this number is made up and we'll get conflicts one days

Reply:
These values are now moved to RGMBBase.h file